components/style/selector_map.rs

```diff
@@ -226,7 +226,7 @@ impl SelectorMap<Rule> {
             }
         });
 
-        if let Some(rules) = self.local_name_hash.get(rule_hash_target.local_name()) {
+        if let Some(rules) = self.local_name_hash.get(&rule_hash_target.get_local_name()) {
```

Comment:
This method invokes `clone()` now. Which is bad. By I don't know how to fix it.

Reply:
This doesn't need to use `selectors::Element`. You can move the `fn local_name` signature to `TElement`. So `selectors` doesn't need `get_local_name`, and this shouldn't need to call `clone()`.